src/TextInput/index.js

```diff
@@ -79,20 +84,43 @@ class TextInput extends React.Component<Props, State> {
     }));
   };
 
-  onFocus = () => {
+  handleOnFocus = () => {
     const { onFocus, disabled } = this.props;
-    if (!disabled) {
-      this.toggleFocus();
-      onFocus && onFocus();
+    if (disabled) {
+      return;
     }
+
+    this.toggleFocus();
+    onFocus && onFocus();
```

Comment:
Same as `handleChangeText`. This `onFocus && onFocus();` is just weird, it should be at least:

```js
if (onFocus) {
  onFocus();
}
```